### PR TITLE
[BISERVER-12918] the report parameter/prompt panel flashes when autosubmit off

### PR DESCRIPTION
**Pull request by MikhailHubanau**
Changed nullValueParams condition to ignore single components (Text Area, Text Box, etc.).
Added unit test.

@pamval, @diogofscmariano, @krivera-pentaho, @annapopova1 could you please review?

### CODE REVIEW

**Comment by krivera-pentaho** [comment]
The change looks good to me, but I definitely want @annapopova1 to weigh in on this before I merge it.

File: test-js/unit/prompting/PromptPanelSpec.js [hunk 1]
```diff
@@ -394,16 +394,54 @@ define([ 'dojo/number', 'dojo/i18n', 'common-ui/prompting/PromptPanel',
...
+
+          panel.parameterChanged(param, name);
+          expect(panel.nullValueParams).toBeUndefined();
+          expect(panel.parametersChanged).toBeTruthy();
+          expect(panel._setTimeoutRefreshPrompt).toHaveBeenCalled();
+        });
+
+        it("should not fill nullValueParams for single components (Text Area, Text Box, etc.) with not null values", function() {
+          param.list = false; // means that it is single component
+          var value = "value"
```
Comment (annapopova1) [inline test-js/unit/prompting/PromptPanelSpec.js]: I don't find where "value" variable is used.

File: test-js/unit/prompting/PromptPanelSpec.js [hunk 2]
```diff
@@ -394,16 +394,54 @@ define([ 'dojo/number', 'dojo/i18n', 'common-ui/prompting/PromptPanel',
...
-        var param = {};
-        var name = "name";
-        spyOn(panel, "_setTimeoutRefreshPrompt");
-        panel.parameterChanged(param, name);
-        expect(panel.parametersChanged).toBeTruthy();
-        expect(panel._setTimeoutRefreshPrompt).toHaveBeenCalled();
-        expect(panel.nullValueParams).toBeDefined();
-        expect(panel.nullValueParams.length).toBe(1);
-        expect(panel.nullValueParams[0]).toBe(param);
+      describe("parameterChanged", function(){
```
Comment (annapopova1) [inline test-js/unit/prompting/PromptPanelSpec.js]: Code style: lost space "function(){"

**Comment by annapopova1** [comment]
IMO will be better to have some tests for checking values with "" and "null" because they are also important for "if" statement.

**Comment by MikhailHubanau** [comment]
Done.

**Pull request merged**